### ISSUE TITLE
openpgpjs loaded twice into memory?

### ISSUE THREAD

**Issue by ghost** [openpgpjs/openpgpjs#839]
I'm unsure if this is an issue or not, so please excuse my ignorance if it isn't :)

I have a website that utilizes openpgpjs. It goes something like this:

```
<html>
<body>
<script src="openpgp.min.js"></script>
openpgp.initWorker({ path: "openpgp.worker.min.js" });
</body>
</html>
```

First, openpgp.min.js is loaded into the page itself, providing the initWorker function. It is my impression that the worker script (openpgp.worker.min.js) itself also loads openpgp.min.js -- leading me to believe that openpgp.min.js is now loaded twice into memory. Is this the case, and if so, is there any way to remedy this seemingly excessive memory usage?

Kind regards, Emil

**Comment by twiss** [comment]
Hi Emil! You are correct. Here's some things you could do about this:

1. Not create the worker at all. This causes more work to be done on the main thread, but it could be acceptable especially if most of the functions you use use Web Crypto, which always happens off the main thread (`generateKey` and `encrypt` use Web Crypto to some extent, `decrypt` doesn't unless you enable AEAD, etc., but mostly you should just try it out and see if it's fast enough for your needs.)
2. Create a worker manually, load openpgp.min.js from there, and manually send messages back and forth to the worker.
3. Extract `async_proxy.js` (which provides the `initWorker` functionality) and its dependencies into a separate script, along with shallow functions which delegate to the worker adapted from `src/openpgp.js` (along with new ones for message and key parsing, perhaps).

We would like to do one of 2 or 3 anyway, and perhaps provide a separate library for this, but we don't have anything concretely planned yet. If you'd like to take a stab at it, feel free to.

**Comment by twiss** [comment]
Closed by #964.
